src/transformers/kernels/grounding_dino/cpu/ms_deform_attn_cpu.cpp

```diff
@@ -0,0 +1,40 @@
+/*!
+**************************************************************************************************
```

Comment:
Do we need new kernels for Grounding DINO? Can't we reuse the ones from Deformable DETR? https://github.com/huggingface/transformers/tree/main/src/transformers/kernels/deformable_detr

Reply:
We could, but then the same would be true to `Deta` https://github.com/huggingface/transformers/tree/main/src/transformers/kernels/deta as it's just a copy and paste

Reply:
They were probably duplicated because the kernels used to live at the model level i.e. `src/transformers/models/model_name/` - and there shouldn't be imports from one modeling file to another. 

Happy to have them shared i.e. grounding DINO just points to deformable DETR kernels in the modeling file. 

There's a possible issue where changes might be made to the kernels to e.g. update deformable DETR which isn't compatible with other dependant models. This would effectively be hidden as it's not obvious from the kernel paths here. The likelihood of this happening is so low, I think we can just ignore it for the moment.

Reply:
`GroundingDino` pointing to `DeformableDetr` kernels here https://github.com/huggingface/transformers/pull/30768/commits/bc1f8cbeb408567e32abd99a4f38d9abf623c8d3